docs/spfx/toolchain/implement-ci-cd-with-azure-devops.md

```diff
@@ -0,0 +1,179 @@
+![importing coverage information](../../images/azure-devops-spfx-05.png)
+
+### Bundling the solution
+You first need to bundle your solution in order to get static assets that can be understood by a web browser. For that add another `gulp` task, set the `gulpfile` path, set the `Gulp Tasks` field to bundle and add `--ship` in the `Arguments`.
+![bundling the assets](../../images/azure-devops-spfx-06.png)
+
+### Packaging the solution
+Now that you have static assets, the next step is to package those into a package SharePoint will be able to deploy. Add another `gulp` task, set the `gulpfile` path, set the `Gulp Tasks` field to `package-solution` and add `--ship` in the `Arguments`.
+![packaging the solution](../../images/azure-devops-spfx-07.png)
+
+### Preparing the artifacts
+By default a Azure DevOps build does not retain any files, you have to explicitly tell it to keep the files you will be needing for the release.  
+Add a `Copy Files` task and set the `Contents` to `**\*.sppkg` (the SharePoint Package created with the previous task) and the target folder to `$(build.artifactstagingdirectory)/drop`.
+![grabbing the artifacts](../../images/azure-devops-spfx-08.png)
+
+### Publishing the artifacts
+Now that you have collected all the files needed for deployment in a special artifacts folder, you still need to instruct Azure DevOps to keep these files after the execution of the build. To do so add a `Publish artifacts` task and set the `Path to publish` to `$(build.artifactstagingdirectory)/drop` and the `Artifact name` to `drop`.
+![publishing the artifacts](../../images/azure-devops-spfx-09.png)
+
+
+## Continuous Deployment
+Continuous deployment helps you deploy your package to different environments and keep tracks of which versions are deployed to which environments.
+
+### Creating the Release Definition
+Start by creating a new release definition with an empty template. A release defition is a process model that defines which environments you are working with, as well at the deployment tasks to executes and which artifacts (from the builds) will be used.
+![creating the release definition](../../images/azure-devops-spfx-10.png)
+
+### Linking the build artifact
+Click on `Add an artifact` and select the build definition you previously created, write down the `Source Alias` name you set as it will impact paths for the tasks later on.
+![linking the artifacts](../../images/azure-devops-spfx-11.png)
+
+### Creating the environment
+You can give a name to your environment, set up pre-deployment approvals, artificats filters (deploy only if the build comes from this or that branch) an much more by clicking on the buttons around the environment box or directly on the title.
+![creating the environment](../../images/azure-devops-spfx-12.png)
+
+### Installing node 10
```

Comment:
During this sequence, we're only installing and using the Office 365 CLI, all the SPFX tasks have been run during the build process. And these installations are no kept or shared between processes. Adding a not for supportability.

Reply:
Wouldn't it be better for simplicity and consistency to install the same version of Node throughout the whole process?

Reply:
Consistency vs performance, I'm not sold to any option. Maybe in that case I could rewrite it to instruct v8 in the main content and hint people to use node 10 if they want better performances as long as they are not calling the spfx tasks in the note?

Reply:
Have you measured the performance difference for this particular scenario in using Node v10 vs v8? Given it's a single task, I suspect, that the performance gain will be negligible and not worth the confusion that this could cause.

Reply:
you're right, 4 seconds difference, done. Thanks!